src/theory/bv/bvgauss.cpp

```diff
@@ -0,0 +1,692 @@
+/*********************                                                        */
+/*! \file bvgauss.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Gaussian Elimination preprocessing pass.
+ **
+ ** Simplify a given equation system modulo a (prime) number via Gaussian
+ ** Elimination if possible.
+ **/
+
+#include "theory/bv/bvgauss.h"
+
+#include <iostream>
+#include <stack>
+#include <unordered_map>
+
+#include "theory/bv/theory_bv_utils.h"
+
+using namespace CVC4;
+using namespace std;
+
+namespace CVC4 {
+namespace theory {
+namespace bv {
+
+unsigned BVGaussElim::getMinBwExpr(Node expr)
+{
+  stack<Node> visit;
+  unordered_map<Node, unsigned, NodeHashFunction> visited;
```

Comment:
This is simply a cache for visited nodes when iteratively traversing through the node structure. Doing such traversals iteratively rather than recursively always follows a certain pattern, hence I prefer to always use the same names when using this pattern. The name you proposed does not improve readability. On the contrary, it makes identifying this pattern more difficult.

Reply:
I'll add a comment to make clear what is mapped to what.